### ISSUE TITLE
Fatal: `Sabre\DAV\Exception` Could not rename part file to final file

### ISSUE THREAD

**Issue by maltepost** [nextcloud/groupfolders#1453]
We get "Could not rename part file to final file"-Errors when trying to write files in shared folders in a group folder (group folder/shared folder/...) on two different servers which were updated to version 21.0.1.1. The two backup servers with version 20.0.9 run fine. Any ideas?

```
[webdav] Fatal: Sabre\DAV\Exception: Could not rename part file to final file at <<closure>>

0. /var/www/nextcloud/apps/dav/lib/Connector/Sabre/Directory.php line 156
   OCA\DAV\Connector\Sabre\File->put(null)
1. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 1098
   OCA\DAV\Connector\Sabre\Directory->createFile("210412_CDD Proj ... X", null)
2. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 504
   Sabre\DAV\Server->createFile("Testuser.docume ... X", null, null)
3. /var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
   Sabre\DAV\CorePlugin->httpPut(Sabre\HTTP\Request {}, Sabre\HTTP\Response {})
4. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 472
   Sabre\DAV\Server->emit("method:PUT", [Sabre\HTTP\Requ ... }])
5. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 253
   Sabre\DAV\Server->invokeMethod(Sabre\HTTP\Request {}, Sabre\HTTP\Response {})
6. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 321
   Sabre\DAV\Server->start()
7. /var/www/nextcloud/apps/dav/appinfo/v1/webdav.php line 84
   Sabre\DAV\Server->exec()
8. /var/www/nextcloud/remote.php line 167
   require_once("/var/www/nextcl ... p")

PUT /remote.php/webdav/Testuser.documents/210412_CDD%.DOCX
from 62.84.220.XXX by Testuser at 2021-04-19T10:19:58+00:00
```

**Comment by JacoboDominguez** [comment]
Same problem, it seems an issue with the folder path, the error reports access to the path:
./data/<GROUP FOLDER>/<path>
while in the server group folders are inside the ./data/__groupfolder/<id>/, e.g.
./data/__groupfolder/<group folder id>/<path>

I fixed temporally creating a symbolink link in ./data/ => to the ./data/__groupfolder/<group folder id>
Is this a bug with the code or should this symbolics link be created?. The symbolic link is a temporary fix for me, because group folder name could collide with an username, that must be the reason to host them in the __groupfolder.
Cheers

**Comment by propcoder** [comment]
I met similar issue and reported here: [github.com/nextcloud/server/issues/26649#issue-862649435](https://github.com/nextcloud/server/issues/26649#issue-862649435)

**Comment by variantolog** [comment]
Same problem.  Servers with version 20.0.9 run fine.

**Comment by variantolog** [comment]
P.S.
IMHO, i assume that when forming the full path, instead of the getSourcePath() function from the class GroupMountPoint class, some other function is used.

**Comment by kaberdouch** [comment]
Same problem after migrating to docker and upgrading from 20.0.8 to 21.0.1

**Comment by HLeithner** [comment]
same problem here Nextcloud 21.0.1

**Comment by N4IR0** [comment]
It seems that this bug also exists after upgrading to Nextcloud 20.0.9

**Comment by bob4os** [comment]
Interestingly moving a file from your Nextcloud root folder into your shared subfolder of a groupfolder is possible.
Permissions on the share do not matter.
(PHP 7.4 and Lighttpd)

**Comment by borekon** [comment]
> Same problem. Servers with version 20.0.9 run fine.
> 
> In my case:
> **/mnt/ncdata/__groupfolder/4** - Group folder (with ID 4)
> **/mnt/ncdata/__groupfolder/4/FSFU** - Folder shared for user (located in group folder with ID 4)
> 
> User tries to upload own files in FSFU
> Error in log: _**unable to rename, source directory is not writable : /mnt/ncdata/FSFU**_
> In fact, of course, there is no such folder at all
> 
> My temporary fix: **ln -s /mnt/ncdata/__groupfolder/4/FSFU /mnt/ncdata/FSFU**

Same here, except if i create a file instead of upload it, the file is created in the right folder and no error is shown

**Comment by Ornanovitch** [comment]
It seems to be a duplicate of the recent #1445

This might affect a lot of users...

**Comment by HLeithner** [comment]
Many things have been fixed in 9.0.1 but not this issue (at least for me)

**Comment by bob4os** [comment]
I can see that files like "... .ocTransferId ... .part" are created in the shared folders - only renaming the file after the transfer fails.
The ".part" files remain in these folders.

**Comment by Skywalker-11** [comment]
Assume:
groupfolder GF grants write access to group WG
User A, B are member of group WG
User C is not a member of group WG

- user B tries to upload a file to the group folder GF: :heavy_check_mark: 
- user A shares subdirectory GF/S of the group folder with B; B uploads a file to the shared directory S (not via the GF but the separate share): :x: 
- user A shares subdirectory GF/S of the group folder with B; B uploads a file to the directory GF/S (via the actual GF): :heavy_check_mark:  
- user A shares any directory (incl. root of GF) with user C; C uploads a file to the shared directory: :x:
- user A creates a share link to any directory of GF; B uploads a file to link shared directory: :heavy_check_mark:  
- user A creates a share link to any directory of GF; C uploads a file to link shared directory: :heavy_check_mark:  

:x: creates the `rename part file to final file` error

So it seems to be an issue of incompatibility of groupfolders with user shares and link shares are working fine.

Setup: groupfolders 9.0.1; NC: 21.0.1; LDAP user and group backend: 1.11.0

**Comment by pgassmann** [comment]
Related to 

- Uploading to shared public upload-only Folder fails: Could not rename part file to final file #8942
- Error when trying to upload Files to a File Drop #23543
- (possibly) Cannot modify header information (Sapi.php:132) NC 19.0.2 #22509

**Comment by juliushaertl** [comment]
Bisect showed that https://github.com/nextcloud/server/pull/25568 seems to be causing this with the steps from https://github.com/nextcloud/groupfolders/issues/1453#issuecomment-829211348

**Comment by Ornanovitch** [comment]
> Bisect showed that nextcloud/server#25568 seems to be causing this with the steps from #1453 (comment)

How could this issue be related to the trashbin? The steps above don't mention anything concerning the trash...

**Comment by trieq** [comment]
The issue is still present in 21.0.2. User with access to direct shares are unable to upload files and get the error 'Could not rename part file to final file'

**Comment by juliushaertl** [comment]
The linked fix will become part of 21.0.3 https://github.com/nextcloud/server/pull/27015

**Comment by exander77** [comment]
How to recover from the part file? I can't seem to be able to decrypt it.
